### PR TITLE
added new workflow for deploying to aws

### PR DESCRIPTION
**Pull request by fabiocicerchia**

### CODE REVIEW

**Comment by AlessandroMinoccheri** [comment]
Is it already a draft or we can review this PR @fabiocicerchia ?

**Comment by fabiocicerchia** [comment]
@AlessandroMinoccheri I've tested it locally with [act](https://github.com/nektos/act) and it is now ready ;)

**Review by AlessandroMinoccheri — APPROVED** [review]
LGTM 👍

**Review by JellyBellyDev — APPROVED** [review]
LGTM

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+jobs:
+  deploy:
+
+    runs-on: ubuntu-18.04
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Add AWS CLI
+      run: apt-get update && apt-get install -y awscli
```
Comment (drAlberT) [inline .github/workflows/deploy.yml:17]: the distribution package is usually outdated, consider installing it as recommended in their doc :)

File: Makefile [hunk 2]
```diff
@@ -106,7 +105,7 @@ deploy_prod: .docker_img_deps ## deploy to prod
 		| tee sys/cloudformation/parameters.secrets.prod.json.new; \
         mv sys/cloudformation/parameters.secrets.prod.json sys/cloudformation/parameters.secrets.prod.json.bak; \
         mv sys/cloudformation/parameters.secrets.prod.json.new sys/cloudformation/parameters.secrets.prod.json; \
-	aws --profile=$(AWS_PROFILE) cloudformation create-change-set \
+	aws --profile=$(AWS_PROFILE) cloudformation create-change-set --capabilities CAPABILITY_NAMED_IAM \
 		--stack=poser-ecs \
 		--change-set-name=poser-ecs-$$VER \
 		--template-body=file://$$PWD/sys/cloudformation/stack.yaml \
```
Comment (drAlberT) [inline Makefile:111]: IIRC it should work wit relative path too

File: Makefile [hunk 3]
```diff
@@ -81,7 +81,6 @@ AWS_ACCOUNT_ID ?= $(shell aws sts get-caller-identity --profile=$(AWS_PROFILE) |
 PREVIOUS_TAG=$(shell git ls-remote --tags 2>&1 | awk '{print $$2}' | sort -r | head -n 1 | cut -d "/" -f3)
 
 deploy_prod: .docker_img_deps ## deploy to prod
-	git fetch --all --tags > /dev/null; \
 	aws ecr get-login-password --profile $(AWS_PROFILE) | docker login --password-stdin -u AWS $(AWS_ACCOUNT_ID).dkr.ecr.eu-west-1.amazonaws.com; \
```
Comment (drAlberT) [inline Makefile:84]: hardcode region, you can use the ENV variable

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+    - name: Configure AWS CLI
+      run: |
+        aws --profile poser configure set region $AWS_REGION
```
Comment (drAlberT) [inline .github/workflows/deploy.yml:21]: I'm wondering why this is needed, as the SDK should be able to get the values from the ENV variables directly ..

File: sys/cloudformation/stack.yaml [hunk 5]
```diff
@@ -547,6 +547,47 @@ Resources:
...
+          Value: !Ref ServiceName
+      UserName: github_action_deploy
+
+  # IAM ACCESS-KEY
+  iamkey:
+    Type: AWS::IAM::AccessKey
+    DependsOn:
+      - iamusergithubactions
+    Properties:
+      UserName: github_action_deploy
```
Comment (drAlberT) [inline sys/cloudformation/stack.yaml:590]:
```
  iamkey:
    Type: AWS::IAM::AccessKey
    Properties:
      UserName: !Ref iamusergithubactions
```

should work

**Review by drAlberT — CHANGES_REQUESTED** [review]

**Pull request merged**